File: lib/swig/java/TypeListComboBox.java
/*
 * Copyright (c) 2013-2014 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

package nc.PKMN;

import javax.swing.*;

public class TypeListComboBox extends JComboBox<String>
{
    public TypeListComboBox(int gen)
    {
        super();

        StringVector typeVec = new StringVector();
        Lists.getTypeList(typeVec, gen);
        for(int i = 0; i < typeVec.size(); i++)
        {
            addItem(typeVec.get(i));
        }

        setEditable(false);
    }
}

File: lib/swig/cs/TypeListComboBox.cs
/*
 * Copyright (c) 2014 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

using System.Windows.Forms;

namespace PKMN
{
    namespace Forms
    {
        public class TypeListComboBox: ComboBox
        {
            public TypeListComboBox(uint gen): base()
            {
                StringVector type_vec = new StringVector();
                Lists.getTypeList(type_vec, gen);
                for(int i = 0; i < type_vec.Count; i++)
                {
                    Items.Add(type_vec[i]);
                }
            }
        }
    }
}
